crates/holochain_types/src/test_utils/chain.rs

```diff
@@ -1,12 +1,100 @@
+//! Implements TestChainItem, a type used with isotest
+
 use std::ops::Range;
 
 use arbitrary::Arbitrary;
 use arbitrary::Unstructured;
 use holo_hash::*;
 use holochain_zome_types::*;
 
-use super::TestChainHash;
-use super::TestChainItem;
+use crate::prelude::ChainItem;
+
+/// The hash type for a [`TestChainItem`]
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    derive_more::From,
+    derive_more::Deref,
+    derive_more::Into,
+)]
+pub struct TestChainHash(pub u32);
+
+impl From<u8> for TestChainHash {
+    fn from(u: u8) -> Self {
+        Self(u as u32)
+    }
+}
+
+impl From<i32> for TestChainHash {
+    fn from(u: i32) -> Self {
+        Self(u as u32)
+    }
+}
+
+isotest::iso! {
+    TestChainHash => |h| hash_from_u32(*h),
+    ActionHash => |h| Self(u32::from_le_bytes(h.get_raw_32()[0..4].try_into().unwrap())),
+    test_cases: [
```

Comment:
i wonder whether calling this `mock_cases` would be more accurate in distinguishing these cases from the _real_ ones which are tested as well

Reply:
Sure I could make this clearer:

`test_cases` is a collection of test data to be tested to make sure it satisfies the conversion invariants
`real_cases` is a collection of real data to be tested to make sure it satisfies the conversion invariants

Reply:
in my understanding it's not "test" vs "real", rather it's "mock" vs "real", because all of this is in a testing context. the extended description would be "mock test cases" and "real test cases".